packages/astro/src/core/build/static-build.ts

```diff
@@ -232,10 +257,17 @@ async function generatePage(output: OutputChunk, opts: StaticBuildOptions, inter
 		renderers,
 	};
 
-	const renderPromises = pageData.paths.map((path) => {
-		return generatePath(path, opts, generationOptions);
-	});
-	return await Promise.all(renderPromises);
+	const renderPromises = [];
+	// Throttle the paths to avoid overloading the CPU with too many tasks.
+	for(const paths of throttle(MAX_CONCURRENT_RENDERS, pageData.paths)) {
+		for(const path of paths) {
+			renderPromises.push(generatePath(path, opts, generationOptions));
+		}
+		// This blocks generating more paths until these 10 complete.
+		await Promise.all(renderPromises);
+		// This empties the array without allocating a new one.
```

Comment:
So this is a form of chunking the builds?

Reply:
The rendering part, yes.